runtime/tr.source/trj9/optimizer/UnsafeFastPath.cpp

```diff
@@ -149,6 +149,16 @@ int32_t TR_UnsafeFastPath::perform()
          bool isVolatile = false;
          bool isArrayOperation = false;
          bool isByIndex = false;
+         int32_t objectChild = 1;
+         int32_t offsetChild = 2;
+
+         switch (symbol->getRecognizedMethod())
+            {
+            case TR::java_lang_StringUTF16_getChar:
+               objectChild = 0;
+               offsetChild = 1;
+               break;
+            }
```

Comment:
Would it be cleaner if this switch was folded into the `TR::java_lang_StringUTF16_getChar` case in the below switch table? i.e. you can do:

```
             case TR::java_lang_StringUTF16_getChar:
                objectChild = 0;
                offsetChild = 1;
             case TR::com_ibm_jit_JITHelpers_getByteFromArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_getByteFromArrayByIndex:
             case TR::com_ibm_jit_JITHelpers_getByteFromArray:
             case TR::com_ibm_jit_JITHelpers_getCharFromArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_getCharFromArrayByIndex:
             case TR::com_ibm_jit_JITHelpers_getCharFromArray:
             case TR::com_ibm_jit_JITHelpers_getIntFromArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_getIntFromArray:
             case TR::com_ibm_jit_JITHelpers_getLongFromArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_getLongFromArray:
             case TR::com_ibm_jit_JITHelpers_getObjectFromArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_getObjectFromArray:
             case TR::com_ibm_jit_JITHelpers_putByteInArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_putByteInArrayByIndex:
             case TR::com_ibm_jit_JITHelpers_putByteInArray:
             case TR::com_ibm_jit_JITHelpers_putCharInArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_putCharInArrayByIndex:
             case TR::com_ibm_jit_JITHelpers_putCharInArray:
             case TR::com_ibm_jit_JITHelpers_putIntInArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_putIntInArray:
             case TR::com_ibm_jit_JITHelpers_putLongInArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_putLongInArray:
             case TR::com_ibm_jit_JITHelpers_putObjectInArrayVolatile:
             case TR::com_ibm_jit_JITHelpers_putObjectInArray:
                isArrayOperation = true;
                break;
```

Reply:
I created a separate switch because there may well be other cases where we want to change the offsets that are not related to arrayness. The basic pattern here is we have a switch to set a particular flag or set of values and I think I prefer the idea of doing the object and offset in their own so it is clear how things map. It may make an extra lookup table, but the cost should be low in terms of compile time and it makes the logic more clear in my opinion.